src/main.cpp

```diff
@@ -692,7 +691,7 @@ unsigned int GetP2SHSigOpCount(const CTransaction& tx, const CCoinsViewCache& in
 
 
 
-bool CheckTransaction(const CTransaction& tx, CValidationState &state)
+bool CheckTransaction(const CTransaction& tx, CValidationState &state, uint64_t nMaxTxSize)
```

Comment:
I left this as nMaxTxSize for now because it helps with testing (sometimes you want to test a transaction and ignore the size constraint, so you set it to -1). We could always change this to params later if needed.